src/blt_mad/helpers.go

```diff
@@ -146,26 +134,96 @@ func withinTolerance(a float64, b float64, e float64) bool {
 	}
 }
 
-func withinToleranceFloatSlice(a []float64, b []float64, e float64) bool {
+func WithinToleranceFloatSlice(a []float64, b []float64, e float64) bool {
 	if reflect.DeepEqual(a, b) {
 		return true
 	} else {
 		for i := 0; i < len(a); i++ {
-			if !withinTolerance(a[i], b[i], e) {
+			if !WithinTolerance(a[i], b[i], e) {
 				return false
 			}
 		}
 	}
 	return true
 }
 
-func readTxtToString(filePath string) (string, error) {
-	// Read the entire file into a byte slice
-	content, err := ioutil.ReadFile(filePath)
+func calculatePercentile(numbers []float64, percentile float64) float64 {
+	sort.Float64s(numbers)
```

Comment:
Are we okay with in-place sorting of `numbers`? Or should the function work with a copy of its input?